### PR TITLE
Build capella from source

### PR DESCRIPTION
**Pull request by MoritzWeber0**
Adds a script + Docker image + README section to build Capella from source (including `aarch64` support).

### CODE REVIEW

File: builder/build_capella_from_source.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# Inject JRE 17 (macOS)
+curl -L -o macJRE-aarch64.tar.gz https://api.adoptium.net/v3/binary/latest/17/ga/mac/aarch64/jre/hotspot/normal/eclipse?project=jdk
+mkdir -p macJRE-aarch64/jre
+tar -xf macJRE-aarch64.tar.gz --strip-components 1 -C macJRE-aarch64/jre
+
+# Inject JRE 17 (Windows)
+curl -L -o winJRE.zip https://api.adoptium.net/v3/binary/latest/17/ga/windows/x64/jre/hotspot/normal/eclipse?project=jdk
+mkdir -p winJRE
+unzip winJRE.zip -d winJRE/jre
+mv winJRE/jre/*/* winJRE/jre
```
Comment (amolenaar) [inline builder/build_capella_from_source.sh:65]: Does it make sense to include the windows JRE? aarch64 is not supported on windows, right?

**Review by amolenaar — APPROVED** [review]
LGTM

I did not run the scripts myself, though. I suppose it only runs on aarch64 machines.

File: builder/build_capella_from_source.sh (hunk 1 above)
Comment (MoritzWeber0) [reply builder/build_capella_from_source.sh:65]: The script runs on all supported architectures and also builds for `x86_64`. Windows is only built for x86_64.

File: README.md [hunk 2]
```diff
@@ -714,6 +714,25 @@ We have explicitly observed the following:
...
+
+```
+CAPELLA_VERSION=6.0.0 make capella/builder
+```
+
+When the script is successful, you should see the result in `builder/output`.
+
+We store a cache of the maven repository in the `builder/m2_cache` directory.
+If you don't need the cache anymore, you can delete the directory after building.
+
```
Comment (jamilraichouni) [inline README.md:735]: How can I limit the build to one or a set of platforms?
I just ran a build and got

```text
capella-6.0.0.202302172253-linux-gtk-aarch64.tar.gz
capella-6.0.0.202302172253-linux-gtk-x86_64.tar.gz
capella-6.0.0.202302172253-macosx-cocoa-aarch64.tar.gz
capella-6.0.0.202302172253-macosx-cocoa-x86_64.tar.gz
capella-6.0.0.202302172253-win32-win32-x86_64.zip  
```
That took 35 minutes and I was just interested in one build.

Comment (MoritzWeber0) [reply README.md:735]: There is no way to limit it - and there is also not a significant speed improvement when building against one platform. There are just a few small platform specific dependencies which have to be pulled for each platform individually - all additional dependencies are platform independent. 

Therefore, I decided that it doesn't really make sense to just build against one platform - if you don't need the other platforms, just delete the files.

Comment (jamilraichouni) [reply README.md:735]: Ah, okay, thanks! Hoped that one could save a lot of build time.

**Pull request merged**